src/Framework/Constraint/IsResourceType.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\Framework\Constraint;
+
+/**
+ * Constraint that checks if value is resource type
+ *
+ * The file path to check is passed as $other in evaluate().
+ */
+class IsResourceType extends Constraint
+{
+    /**
+     * @var string
+     */
+    private $type;
+
+    public function __construct($type)
```

Comment:
i know it's picky, but I could pass non-string here as well (eg, file descriptor ;P)
then, the behaviour could be funny ;)

Reply:
You are totally right.
Fixed.